### PR TITLE
Allow fragment spreads on union types.

### PR DESCRIPTION
**Pull request by Bellk17**
Instead of doing an inline fragment (... on SocialAccount {...}), can now use a fragment (...socialAccountFragment). Fragments generate type definitions, so are much more useful when parsing / propagating more complex responses.

### CODE REVIEW

File: test/test1.graphql [hunk 1]
```diff
@@ -1,12 +0,0 @@
-query q1
```
Comment (AsaAyers) [inline test/test1.graphql:1]: Was this file supposed to be removed?

Comment (jxbadam) [reply test/test1.graphql:1]: Well, was it @Bellk17 ? :)

Comment (Bellk17) [reply test/test1.graphql:1]: Original author had written a test query, but never implemented any tests.

**Review by jxbadam — APPROVED** [review]
:shipit: :fire:

**Pull request merged**